### PR TITLE
Backend: Scheduled Auto Approval Job

### PR DESCRIPTION
**Pull request by shaangill025**
*Issue:*https://github.com/bcgov/entity/issues/22511

*Description of changes:*
Applications will now get picked up after a specified delay in the background job for auto approval processing.

Followup needed - service account token with id `strr-service-account` needs to have `system` role assigned to it.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namex license (Apache 2.0).

### CODE REVIEW

**Comment by shaangill025** [comment]
image

**Review by thorwolpert — COMMENTED** [review]
This looks to me like it assumes the container is always running, which in our case is generally not true.
We use Cloud Scheduler to initiate jobs, which we usually have as CloudRun services.

**Review by dimak1 — APPROVED** [review]
Besides Lekshmi's comments, I have nothing to add. 👍🏻

File: jobs/auto-approval/src/auto_approval/job.py [hunk 1]
```diff
@@ -51,13 +56,38 @@ def shell_context():
...
+    return Application.query.filter(
+        Application.application_date <= cutoff_time,
+        Application.status == Application.Status.PAID,
+    ).all()
+
+
+def process_applications(app, applications):
+    """Process auto-approval for submitted applications."""
+    token = AuthService.get_service_client_token()
+    token_info = decode_token(token)
```
[the remark this answers is not in the document]
Comment (shaangill025) [reply jobs/auto-approval/src/auto_approval/job.py:74]: We need the `jwt_oidc_token_info` when calling the `generate_registration_certificate` function. Cannot rely on `flask_jwt_oidc` as it assigns to `g.jwt_oidc_token_info` within the request context [not accessible when calling `process_auto_approval` from the job].

File: strr-api/src/strr_api/services/approval_service.py [hunk 2]
```diff
@@ -196,6 +197,16 @@ def process_auto_approval(cls, token, application: Application):
                             owner_title_match = False
                         if owner_title_match:
                             auto_approval.title_check = True
+                            registration = RegistrationService.create_registration(
```
[the remark this answers is not in the document]
Comment (shaangill025) [reply strr-api/src/strr_api/services/approval_service.py:208]: Done

**Pull request merged**